source/scid/linalg.d

```diff
@@ -418,10 +418,22 @@ body
 
 template DetType(MatrixT)
 {
-    static if (scid.core.traits.isComplex!(BaseElementType!MatrixT))
```

Comment:
I agree with this change, and I actually think we can take it even further: How about making `det()` always return `BaseElementType!MatrixT`, and removing the `DetType` template altogether? Returning `real` from a function which takes a `float` or `double` matrix is bound to confuse people, especially if the return type on x86 is different from other platforms.